commit a5e0ace04fbf56c1794b1a2fa7a93672753b3fc7
Author: Gustavo A. R. Silva <gustavoars@kernel.org>
Date:   Thu Nov 30 14:29:34 2023 -0600

    init: Kconfig: Disable -Wstringop-overflow for GCC-11
    
    -Wstringop-overflow is buggy in GCC-11. Therefore, we should disable
    this option specifically for that compiler version. To achieve this,
    we introduce a new configuration option: GCC11_NO_STRINGOP_OVERFLOW.
    
    The compiler option related to string operation overflow is now managed
    under configuration CC_STRINGOP_OVERFLOW. This option is enabled by
    default for all other versions of GCC that support it.
    
    Link: https://lore.kernel.org/lkml/b3c99290-40bc-426f-b3d2-1aa903f95c4e@embeddedor.com/
    Link: https://lore.kernel.org/lkml/20231128091351.2bfb38dd@canb.auug.org.au/
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/linux-hardening/ZWj1+jkweEDWbmAR@work/
    Signed-off-by: Gustavo A. R. Silva <gustavoars@kernel.org>

diff --git a/Makefile b/Makefile
index c01cbbb208ca..9f9b76d3a4b7 100644
--- a/Makefile
+++ b/Makefile
@@ -986,7 +986,9 @@ NOSTDINC_FLAGS += -nostdinc
 # perform bounds checking.
 KBUILD_CFLAGS += $(call cc-option, -fstrict-flex-arrays=3)
 
-KBUILD_CFLAGS += $(call cc-option, -Wstringop-overflow)
+#Currently, disable -Wstringop-overflow for GCC 11, globally.
+KBUILD_CFLAGS-$(CONFIG_CC_NO_STRINGOP_OVERFLOW) += $(call cc-option, -Wno-stringop-overflow)
+KBUILD_CFLAGS-$(CONFIG_CC_STRINGOP_OVERFLOW) += $(call cc-option, -Wstringop-overflow)
 
 # disable invalid "can't wrap" optimizations for signed / pointers
 KBUILD_CFLAGS	+= -fno-strict-overflow